pkg/apis/servicecatalog/validation/instance.go

```diff
@@ -309,3 +300,62 @@ func ValidateServiceInstanceReferencesUpdate(new *sc.ServiceInstance, old *sc.Se
 	allErrs = append(allErrs, internalValidateServiceInstance(new, false)...)
 	return allErrs
 }
+
+func validatePlanReference(p *sc.PlanReference, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Just to make reading of the conditionals in the code easier.
+	externalClassSet := p.ExternalClusterServiceClassName != ""
+	externalPlanSet := p.ExternalClusterServicePlanName != ""
+	k8sClassSet := p.ClusterServiceClassName != ""
+	k8sPlanSet := p.ClusterServicePlanName != ""
+
+	// Can't specify both External and k8s name but must specify one.
+	if externalClassSet == k8sClassSet {
+		allErrs = append(allErrs, field.Required(fldPath.Child("externalClusterServiceClassName"), "exactly one of externalClusterServiceClassName or clusterServiceClassName required"))
+		allErrs = append(allErrs, field.Required(fldPath.Child("clusterServiceClassName"), "exactly one of externalClusterServiceClassName or clusterServiceClassName required"))
+	}
+	// Can't specify both External and k8s name but must specify one.
+	if externalPlanSet == k8sPlanSet {
+		allErrs = append(allErrs, field.Required(fldPath.Child("externalClusterServicePlanName"), "either externalClusterServicePlanName or clusterServicePlanName required"))
```

Comment:
Nit: This one's message is different than the other three. ("either" vs "exactly one of")